src/rust/engine/client/src/options/mod.rs

```diff
@@ -0,0 +1,223 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+mod args;
+mod config;
+mod env;
+mod id;
+mod parse;
+
+use std::collections::{BTreeMap, HashSet};
+use std::ops::Deref;
+use std::os::unix::ffi::OsStrExt;
+use std::path::Path;
+use std::rc::Rc;
+
+use crate::build_root::BuildRoot;
+use crate::option_id;
+
+use args::Args;
+use config::Config;
+use env::Env;
+use parse::{parse_bool, parse_string_list};
+
+pub use id::{OptionId, Scope};
+
+#[derive(Copy, Clone, Debug)]
+pub(crate) enum ListEditAction {
+  REPLACE,
+  ADD,
+  REMOVE,
+}
+
+#[derive(Debug)]
+pub(crate) struct ListEdit<T> {
+  pub action: ListEditAction,
+  pub items: Vec<T>,
+}
+
+trait OptionsSource {
```

Comment:
Doc comments for the trait and its methods?

Reply:
Documented.